libglusterfs/src/glusterfs/common-utils.h

```diff
@@ -1245,18 +1245,18 @@ gf_tvdiff(struct timeval *start, struct timeval *end)
 
 /* Return delta value in nanoseconds. */
 
-static inline double
+static inline uint64_t
 gf_tsdiff(struct timespec *start, struct timespec *end)
 {
     struct timespec t;
 
     if (start->tv_nsec > end->tv_nsec)
-        t.tv_sec = end->tv_sec - 1, t.tv_nsec = end->tv_nsec + 1000000000;
+        t.tv_sec = end->tv_sec - 1, t.tv_nsec = end->tv_nsec + 1e9;
     else
         t.tv_sec = end->tv_sec, t.tv_nsec = end->tv_nsec;
```

Comment:
All this conversion is not needed. The returned value will be correct without this.

Reply:
Meaning just setting t.tv_sec = end->tv_sec and t.tv_nsec = end->tv_nsec is enough?

Reply:
You can directly use `end` instead of `t`.

To show it more clearly, suppose e = `end`, s = `start`, x.s = `x.tv_sec` and x.n = `x.tv_nsec`:

    t.s = e.s - 1
    t.n = e.n + 1000000000

Then:

    (t.s - s.s) * 1000000000 + (t.n - s.n) =
        ((e.s - 1) - s.s) * 1000000000 + ((e.n + 1000000000) - s.n) =
        (e.s - s.s - 1) * 1000000000 + (e.n - s.n + 1000000000) =
        (e.s - s.s) * 1000000000 - 1000000000 + (e.n - s.n) + 1000000000 =
        (e.s - s.s) * 1000000000 + (e.n - s.n)

So the result is the same without modifying `end` and `start`.